include.sh

```diff
@@ -0,0 +1,130 @@
+export ENROOT_DATA_PATH=${CI_WS}/ENROOT_DATA
+export SLURM_IDS_PATH=${CI_WS}/SLURM_IDS
+
+
+# Set a variable CONTAINER_NAME based on the environment variable
+# CUSTOM_ENV_USE_NAME
+if [[ -z "${CUSTOM_ENV_USE_NAME}" ]]; then
+    CONTAINER_NAME="GitLabRunnerCustomExecutorEnrootBuildID${CUSTOM_ENV_CI_BUILD_ID}"
+else
+    CONTAINER_NAME="${CUSTOM_ENV_USE_NAME}"
+fi
+export CONTAINER_NAME
+
+
+# CCACHE and volume management
+export CCACHE_DIR_SRC="${CI_WS}/ccache"
+ENROOT_MOUNT_OPTIONS=()
+if [[ -n "${CUSTOM_ENV_CCACHE_DIR}" ]]; then
+    if [[ ! -d "${CCACHE_DIR_SRC}" ]]; then
+        mkdir -p "${CCACHE_DIR_SRC}"
+    fi
+    ENROOT_MOUNT_OPTIONS+=(--mount "${CCACHE_DIR_SRC}:${CUSTOM_ENV_CCACHE_DIR}" )
+fi
+export ENROOT_MOUNT_OPTIONS
+
+
+# Propagate these environment variables to the container
+PROPAGATED_ENV_VARIABLES=(BENCHMARK
+                          DRY_RUN
+                          EXECUTOR
+                          REPETITIONS
+                          SOLVER_REPETITIONS
+                          SEGMENTS
+                          SEGMENT_ID
+                          PRECONDS
+                          FORMATS
+                          ELL_IMBALANCE_LIMIT
+                          SOLVERS
+                          SOLVERS_PRECISION
+                          SOLVERS_MAX_ITERATIONS
+                          SOLVERS_GMRES_RESTART
+                          SYSTEM_NAME
+                          DEVICE_ID
+                          SOLVERS_JACOBI_MAX_BS
+                          BENCHMARK_PRECISION
+                          SOLVERS_RHS
+                          SOLVERS_RHS_FLAG
+                          SOLVERS_INITIAL_GUESS
+                          GPU_TIMER
+                          DETAILED
+                          MATRIX_LIST_FILE
+                          NVIDIA_VISIBLE_DEVICES
+                          CCACHE_DIR
+                          CCACHE_MAXSIZE
+                         )
+for bench_var in ${PROPAGATED_ENV_VARIABLES[*]}; do
+    check_var="CUSTOM_ENV_${bench_var}"
+    if [[ -n "${!check_var}" ]]; then
+        ENROOT_ENV_CONFIG+=("-e ${bench_var}=${!check_var}")
+    fi
+done
+export ENROOT_ENV_CONFIG
+
+# SLURM configuration variables.
+#
+# If the user sets any slurm variable or the variable USE_SLURM, this container
+# will use slurm job submission
+SUPPORTED_SLURM_VARIABLES=(SLURM_PARTITION
+                           SLURM_EXCLUSIVE
+                           SLURM_TIME
+                           SLURM_GRES
+                           SLURM_UPDATE_INTERVAL
+                           SLURM_PENDING_LIMIT
+                           SLURM_RUNNING_LIMIT
+                           USE_SLURM)
+for slurm_var in ${SUPPORTED_SLURM_VARIABLES[*]}; do
+    check_var="CUSTOM_ENV_${slurm_var}"
+    if [[ -n "${!check_var}" ]]; then
+        USE_SLURM=1
+    fi
+done
+export USE_SLURM
+# variables from slurm_utils we need to expose outside
+export SLURM_UPDATE_INTERVAL
```

Comment:
some of SUPPRTED_SLURM_VARIABLE are not exported?

Reply:
They are used directly in the form `CUSTOM_ENV_<SLURM_VARIABLE>` as they need to be processed separately in `run.sh` for building the `SLURM_CONFIG` array.